compiler/src/iree/compiler/Dialect/Stream/Transforms/SpecializeEncodings.cpp

```diff
@@ -134,6 +134,34 @@ static LogicalResult addLayoutsToTensorPhaseOps(
               });
               return success();
             })
+            .Case<IREE::Stream::TensorEmptyOp,
```

Comment:
So... to handle `TensorSizeOfOp` is basically the same as `TensorEmptyOp` and `TensorSplatOp`? You can just combine the 3 in a single case.

Reply:
They have different parameter name. One uses encoding, and the other uses result_encoding. There are other operations use target_encoding. That's why I can't combine all of them together.

Reply:
ahh, then perhaps one way to do it is to refactor them into a static function. 

Generally speaking less duplication is better but this particular case is up to you.

Reply:
I refactor them out because the function is becoming larger and larger. Thanks for pointing it out! For the result_encoding one, I use a template, so we still don't have dup code.